proto/opamp.proto

```diff
@@ -798,6 +807,21 @@ message RemoteConfigStatus {
     string error_message = 3;
 }
 
+// Status: [Development]
+message ConnectionSettingsStatus {
+    // The hash of the connection settings that was last recieved by this Agent
+    // in the connection_settings.hash field. The Server SHOULD compare this
+    // hash with the OfferedConnectionSettings hash it has for the Agent and if
+    // the hashes are different the Server MUST include the connection_settings
+    // field in the response in the ServerToAgent message.
+    bytes last_connection_settings_hash = 1;
+
+    RemoteConfigStatuses status = 2;
```

Comment:
I think it is confusing to have connection setting status reflected via a remote config status enum.

We probably have a couple options:
1. Create a copy of RemoteConfigStatuses enum and call it ConnectionSettingsStatuses.
2. Rename RemoteConfigStatuses to some sort of "Generic" Statuses enum (not sure what's a good name for it). This only works if we are certain the statuses for both of these things are always going to require the same enum set. I haven't thought through it, so not sure if it is a great idea.

Reply:
@andykellr do you have any opinions on the direction?

I'm leaning towards option 1 as I also can't think of a good rename for option 2.